### PR TITLE
Fix IHaskell convert for ipynb files with nbformat 4.

### PR DESCRIPTION
**Pull request by juselius**
``IHaskell convert`` failed to to convert  .ipynb files with nbformat 4 to lhs. The back transformation produced v3 files, which could subsequently be converted to v4 upon loading into the notebook. These patches fix the issues, making ``convert`` operate on v4 files only.

### CODE REVIEW

**Comment by juselius** [comment]
I was a bit optimistic. lhs to ipynb does not quite work yet.

File: src/IHaskell/Convert/LhsToIpynb.hs [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE CPP #-}
```
Comment (gibiansky) [inline src/IHaskell/Convert/LhsToIpynb.hs]: What do we need CPP for?

**Comment by gibiansky** [comment]
Cool. Let me know when lhs to ipynbV4 is ready to merge.

I would also love to have some tests of the functionality in `Hspec.hs`, as that is now actually checked with Travis CI, but I'm also happy to just file that as an issue once this is merged and add it later.

File: src/IHaskell/Convert/LhsToIpynb.hs (hunk 1 above)
Comment (juselius) [inline src/IHaskell/Convert/LhsToIpynb.hs]: VERSION_ghc is defined as a cpp macro. I'll have a fully working version soon.

Comment (gibiansky) [inline src/IHaskell/Convert/LhsToIpynb.hs]: Ah, got it! I scanned through the source for `#endif` or other directives but saw nothing; didn't think of the cabal macros.

**Comment by juselius** [comment]
Ok, now everything seems to work!

**Pull request merged**